command.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2016 Keybase, Inc. All rights reserved. Use of
+// this source code is governed by the included BSD license.
+
+package updater
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"time"
+
+	"github.com/keybase/go-logging"
+)
+
+func timerWithTimeout(timeout time.Duration, cmd *exec.Cmd, log logging.Logger) *time.Timer {
+	return time.AfterFunc(timeout, func() {
+		if cmd != nil && cmd.Process != nil {
+			log.Warningf("Command timed out, killing")
+			if err := cmd.Process.Kill(); err != nil {
+				log.Warningf("Error trying to kill process: %s", err)
+			}
+		}
+	})
+}
+
+// RunCommand runs a command and returns the combined stdout/err output
+func RunCommand(command string, args []string, timeout time.Duration, log logging.Logger) (string, error) {
+	out, _, err := runCommand(command, args, timeout, log)
+	return out, err
+}
+
+func runCommand(command string, args []string, timeout time.Duration, log logging.Logger) (string, *os.Process, error) {
+	log.Debugf("Command: %s %s", command, args)
+	cmd := exec.Command(command, args...)
+	if cmd == nil {
+		return "", nil, fmt.Errorf("No command")
+	}
+	timer := timerWithTimeout(timeout, cmd, log)
+	if timer != nil {
+		defer timer.Stop()
+	}
```

Comment:
If timer is nil, it means time.AfterFunc returned nil... The docs  https://golang.org/pkg/time/#AfterFunc don't say if this could happen.

Because that std library method returns a ptr, the check here is preventing a crash if it returns nil.